reference-implementation/lib/writable-stream.js

```diff
@@ -932,26 +947,28 @@ function WritableStreamDefaultControllerGetBackpressure(controller) {
 function WritableStreamDefaultControllerError(controller, e) {
   const stream = controller._controlledWritableStream;
 
-  assert(stream._state === 'writable' || stream._state === 'closing');
-
-  const oldState = stream._state;
+  assert(stream._state === 'writable');
 
   stream._state = 'errored';
   stream._storedError = e;
 
-  if (stream._pendingAbortRequest === undefined && stream._writer !== undefined) {
-    let readyPromiseIsPending = false;
-    if (oldState === 'writable' &&
-        WritableStreamDefaultControllerGetBackpressure(stream._writableStreamController) === true) {
-      readyPromiseIsPending = true;
+  // TODO(tyoshino): Given that writer.abort() fails immediately when the stream has been errored, shouldn't
```

Comment:
I think it would be good if the behaviour was the some regardless of the order of controller.error() and abort(), but it's not sufficiently important to be worth adding additional complexity.

In general I don't feel strongly about what happens after controller.error() because I assume it means the underlying transport has failed and any ongoing writes are just going to be lost anyway.